### PR TITLE
(DOCSP-24501): Flutter API key auth

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Document Realm Flutter SDK API key authentication

### Jira

- https://jira.mongodb.org/browse/DOCSP-24501

### Staged Changes (Requires MongoDB Corp SSO)

- [Authenticate Users (Flutter), API Key User section](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-24501/sdk/flutter/users/authenticate/#api-key-user)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/examples/generated/flutter/authenticate_users_test.snippet.work-with-api-keys.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+// Get existing user API key by ID
+// Returns `null` if no existing API key for the ID
+ApiKey? refetchedApiKey = await user.apiKeys.fetch(apiKey.id);
+
+// Get all API keys for a user
+List<ApiKey> apiKeys = await user.apiKeys.fetchAll();
+
+// Disable API key
+await user.apiKeys.disable(apiKey.id);
```
Comment (mongomoe) [inline source/examples/generated/flutter/authenticate_users_test.snippet.work-with-api-keys.dart:12]: Earlier you get a variable "apiKeys", but here you don't use that variable. Any reason?

**Review by mongomoe — APPROVED** [review]
LGTM

File: source/examples/generated/flutter/authenticate_users_test.snippet.work-with-api-keys.dart (hunk 1 above)
Comment (mongodben) [reply source/examples/generated/flutter/authenticate_users_test.snippet.work-with-api-keys.dart:12]: yes b/c that variable is a list of api keys gotten with the method `user.apiKeys.fetch()`, but this is the `user.apiKey.disable()` method.

File: source/sdk/flutter/users/authenticate.txt [hunk 2]
```diff
@@ -105,6 +105,32 @@ on a JWT string. Then pass the credential to ``app.logIn``.
...
+
+To log in with API key authentication, create an :flutter-sdk:`ApiKey <realm/ApiKey-class.html>`
+credential by calling :flutter-sdk:`Credentials.apiKey() <realm/Credentials/Credentials.apiKey.html>`
+on an API key string. Then pass the credential to ``app.logIn()``.
+
+.. literalinclude:: /examples/generated/flutter/authenticate_users_test.snippet.api-key-auth.dart
+   :language: dart
+
+To generate a server API key to use in your credentials,
+refer to the :ref:`Create a Server API Key documentation <api-key-authentication-server-api-key>`.
```
Comment (desistefanova) [inline source/sdk/flutter/users/authenticate.txt:124]: It is a little bit confusing what the difference between `server API key` and `user API key` is. Actually they are the same thing, just the permissions are different as I understand.
Maybe it is good to say what the permissions of the server user are. I mean the one that is created with creating the server API key. 
And also we can mention that we can also login with this server user using `Credentials.apiKey(serverApiKey)`.

Comment (mongodben) [reply source/sdk/flutter/users/authenticate.txt:124]: yea that's my understanding as well, though i'm not sure exactly what the permissions are for each. i think first we need better docs for this on the App Services side. 

i've just created a separate ticket to deal with updating the App Services docs and applying to Realm SDKs as well - https://jira.mongodb.org/browse/DOCSP-26045

**Pull request merged**